phpstan.neon

```diff
@@ -10,6 +10,18 @@ parameters:
     message: "#^Left side of && is always true.$#"
     count: 1
     path: lib/Loop/Loop.php
+  -
+    message: "#^Variable \\$timer might not be defined.$#"
+    count: 1
+    path: lib/Loop/Loop.php
+  -
+    message: "#^Only booleans are allowed in an if condition, int<0, max>|false given.$#"
+    count: 1
+    path: lib/Loop/Loop.php
+  -
+    message: "#^Instanceof between Generator<mixed, mixed, mixed, TReturn> and Generator will always evaluate to true.$#"
```

Comment:
it might make sense to [`treatPhpDocTypesAsCertain: false`](https://phpstan.org/config-reference#treatphpdoctypesascertain) since we have defensive code which makes double sure that types in phpdocs are really passed as expected

Reply:
I tried that locally, and it does not make any difference for this message.
Somehow PHPstan "knows" about Promises and Generator and still thinks that the check can never fail.

For now, IMO we can just keep this in ignoreErrors